### PR TITLE
creates ability to add gift messaging within QL

### PR DESCRIPTION
**Pull request by sherryson**
for issue 1266 in glossier-v2
@aamyot can you CR please?

### CODE REVIEW

File: lib/hushed/documents/request/shipment_order.rb [hunk 1]
```diff
@@ -39,7 +39,10 @@ def to_xml
 
                 xml.Extension @shipment.order.number
 
-                xml.Comments @shipment.order.special_instructions
+                xml.Gift @shipment.order.gift.present?
+                xml.SONoteType('GIFTFROM'   => @shipment.order.gift.from)
+                xml.SONoteType('GIFTTO'     => @shipment.order.gift.to)
+                xml.Comments @shipment.order.gift.message
```
Comment (aamyot) [inline lib/hushed/documents/request/shipment_order.rb]: mmmh... what happen if gift is nil?  don't we want to do something like:
```
if @shipment.order.gift.present?
  xml.Gift @shipment.order.gift.present?
  xml.SONoteType('GIFTFROM'   => @shipment.order.gift.from)
  xml.SONoteType('GIFTTO'     => @shipment.order.gift.to)
  xml.Comments @shipment.order.gift.message
end
```

File: spec/unit/documents/request/shipment_order_spec.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module Hushed
   module Documents
     module Request
       describe "ShipmentOrder" do
-        include Hushed::Documents::DocumentInterfaceTestcases
+        # include Hushed::Documents::DocumentInterfaceTestcases
```
Comment (aamyot) [inline spec/unit/documents/request/shipment_order_spec.rb]: why the comment?  if it's useless, can we just delete it.

File: spec/unit/documents/request/shipment_order_spec.rb [hunk 3]
```diff
@@ -33,7 +33,10 @@ module Request
           assert_equal @client.business_unit, document.css('BusinessUnit').first.text
 
           assert_header(document.css('OrderHeader').first)
-          assert_equal @order.note, document.css('Comments').first.text
+          assert_equal @order.gift.present?.to_s, document.css('Gift').first.text
+          assert_equal @order.gift.from, document.css('SONoteType')[0]['GIFTFROM']
+          assert_equal @order.gift.to, document.css('SONoteType')[1]['GIFTTO']
+          assert_equal @order.gift.message, document.css('Comments').first.text
```
Comment (aamyot) [inline spec/unit/documents/request/shipment_order_spec.rb]: we probably want to add a test for when gift is nil

File: spec/unit/documents/request/shipment_order_spec.rb (hunk 2 above)
Comment (sherryson) [inline spec/unit/documents/request/shipment_order_spec.rb]: i'm actually not sure what this is... tests wouldn't run with this line. should i just delete it?

File: lib/hushed/documents/request/shipment_order.rb (hunk 1 above)
Comment (sherryson) [inline lib/hushed/documents/request/shipment_order.rb]: yes good point!

**Comment by aamyot** [comment]
woohoo !  👍
